Consolidated 7.0 Release Notes. (#5027)
apple/foundationdb: documentation/sphinx/source/release-notes/release-notes-700.rst

* Added a new API in all bindings that can be used to get a list of split points that will split the given range into (roughly) equally sized chunks. `(PR #3394) <https://github.com/apple/foundationdb/pull/3394>`_
* Introduced a new role called GRV proxy specialized for serving GRV requests to decrease GRV tail latency since we prioritize commit paths over GRV in the current proxy. The original proxy is renamed to Commit proxy. `(PR #3549) <https://github.com/apple/foundationdb/pull/3549>`_ `(PR #3772) <https://github.com/apple/foundationdb/pull/3772>`_
* Added support for writing backup files directly to Azure blob storage. This is not yet performance tested on large-scale clusters. `(PR #3961) <https://github.com/apple/foundationdb/pull/3961>`_
* Tag-based throttling now also takes the write path into account. `(PR #3512) <https://github.com/apple/foundationdb/pull/3512>`_
* Added the ability to ratekeeper to throttle certain types of tags based on write hot spots in addition to read hot spots. `(PR #3571) <https://github.com/apple/foundationdb/pull/3571>`_

> * Users now have the option to make ratekeeper recommend which transaction tags should be throttled, but not actually throttle them using fdbcli . `(PR #3669) <https://github.com/apple/foundationdb/pull/3669>`_

Reviewer: ```suggestion
* Users now have the option to make ratekeeper recommend which transaction tags should be throttled, but not actually throttle them using fdbcli. `(PR #3669) <https://github.com/apple/foundationdb/pull/3669>`_
```

* Added a new ``--build_flags option`` to binaries to print build information. `(PR #3769) <https://github.com/apple/foundationdb/pull/3769>`_
* Added ``--incremental`` option to backup and restore that allows specification of only recording mutation log files and not range files. Incremental restore also allows restoring to a non-empty destination database. `(PR #3676) <https://github.com/apple/foundationdb/pull/3676>`_

> * Added request tracing framework. `(PR #3329) <https://github.com/apple/foundationdb/pull/3329>`_ `(PR #3700) <https://github.com/apple/foundationdb/pull/3700>`_ `(PR #4048) <https://github.com/apple/foundationdb/pull/4048>`_ `(PR #4125) <https://github.com/apple/foundationdb/pull/4125>`_ `(PR #4204) <https://github.com/apple/foundationdb/pull/4204>`_

Reviewer: ```suggestion
* Added a request tracing framework which breaks down how much time a user request spends in each FDB component. `(PR #3329) <https://github.com/apple/foundationdb/pull/3329>`_ `(PR #3700) <https://github.com/apple/foundationdb/pull/3700>`_ `(PR #4048) <https://github.com/apple/foundationdb/pull/4048>`_ `(PR #4125) <https://github.com/apple/foundationdb/pull/4125>`_ `(PR #4204) <https://github.com/apple/foundationdb/pull/4204>`_
```

> * Added the `(Global Configuration Framework) <https://github.com/apple/foundationdb/wiki/Global-Configuration-Framework>`_, an eventually consistent configuration mechanism to efficiently make runtime changes to all clients and servers. `(PR #4330) <https://github.com/apple/foundationdb/pull/4330>`_

Reviewer 2: This documentation link can be updated to the new global configuration page (`global-configuration.rst`) in #5068.

Author: ok.

> * Added the feature to monitor and manage an fdb cluster via read/write specific special keys through transactions. See :ref:`Documentation <special-keys>` for details. `(PR #3455) <https://github.com/apple/foundationdb/pull/3455>`_

Reviewer: ```suggestion
* Added the ability to monitor and manage an fdb cluster via read/write specific special keys through transactions. See :ref:`Documentation <special-keys>` for details. `(PR #3455) <https://github.com/apple/foundationdb/pull/3455>`_
```

Reliability
-----------
* Backup agents now pause themselves upon a successful snapshot recovery to avoid unintentional data corruption. Operators should manually abort backup agents and clear the backup agent keyspace to avoid using the old cluster's backup configuration. `(PR #4027) <https://github.com/apple/foundationdb/pull/4027>`_
* Log class processes are prioritized above transaction class proceses for becoming tlogs. `(PR #4509) <https://github.com/apple/foundationdb/pull/4509>`_ 
* Improved worker recruitment logic to avoid unnecessary recoveries when processes are added or removed from a cluster. `(PR #4695) <https://github.com/apple/foundationdb/pull/4695>`_ `(PR #4631) <https://github.com/apple/foundationdb/pull/4631>`_ `(PR #4509) <https://github.com/apple/foundationdb/pull/4509>`_

Fixes
-----
* List files asynchronously so many backup files on a slow disk won't cause the backup agent to lose its lease. `(PR #3094) <https://github.com/apple/foundationdb/pull/3094>`_

> * Unknown endpoint has been tracked in a wrong way and therefore showed up too frequently in our statistics. `(PR #4473) <https://github.com/apple/foundationdb/pull/4473>`_

Reviewer: ```suggestion
* Unknown endpoint has been tracked incorrectly and therefore showed up too frequently in our statistics. `(PR #4473) <https://github.com/apple/foundationdb/pull/4473>`_
```
